### PR TITLE
LPS-186772 | Add a testcase to publish a REST API Application

### PR DESCRIPTION
**Pull request by Hijky**
Background:
Add a testcase to publish a REST API Application

Test Plan
CanSeeNewSchemaInPublishedApiApplcationPage
Given an API Application 'App-test' of status ‘published’ by post request postAPIApplication
And Given an API Schema 'AppTest' related to an Application by post request postAPISchema
When I can navigate to App-test page in API Explorer
Then I can see a new schema appear in the page

CanSeeNewApiEndpointInPublishedApiApplcationPage
Given an API Application 'App-test' of status ‘published’ by post request postAPIApplication
And Given an API EndPoint 'AppTest' related to an Application by post request postAPIEndpoint
When I can navigate to App-test page in API Explorer
Then I can see a new endpoint appear in the page without an error

CanSeeNewApiEndpointAndApiSchemaInPublishedApiApplicationPage
Given an API Application 'App-test' of status ‘published’ by post request postAPIApplication
And Given an API Schema 'AppTest' related to an Application by post request postAPISchema
And Given an API EndPoint in an API Application and set Response API Schema to API Endpoints with the created Schema by postAPIEndpoint
When I can navigate to App-test page in API Explorer
Then the page load successfully
And Then I can see a new schema and an endPoint appear in the page

Jira ticket:
https://liferay.atlassian.net/browse/LPS-191033

### CODE REVIEW

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:headless-functional:nocompile

**Comment by Hijky** [comment]
The failures are not related to my PR

**Comment by magjed4289** [comment]
@Hijky do we have a ticket created for the failing test? We not only need to verify whether it is related or not to the PR, but we also need to verify whether the failing test is registered as a failure in Jira for further investigation.

**Comment by Hijky** [comment]
> @Hijky do we have a ticket created for the failing test? We not only need to verify whether it is related or not to the PR, but we also need to verify whether the failing test is registered as a failure in Jira for further investigation. If it's not registered, we need to create a ticket in Jira. If it is registered, it's good to link the Jira ticket in the PR comment.

It's a flaky, I can not reproduce locally

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[APIRest]APIGenerator/PublishARESTAPIApplication.testcase [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+		task ("When I can navigate to App-test page in API Explorer") {
+			APIExplorer.navigateToOpenAPI(customObjectPlural = "app-test");
+		}
+
+		task ("Then the page load successfully") {
+			AssertConsoleTextNotPresent(value1 = "Failed to load API definition.");
+		}
+
+		task ("And Then I can see a new schema and an endPoint appear in the page") {
```
Comment (magjed4289) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[APIRest]APIGenerator/PublishARESTAPIApplication.testcase:64]: not `endPoint` but `endpoint`

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[APIRest]APIGenerator/PublishARESTAPIApplication.testcase [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+		property portal.acceptance = "true";
+
+		task ("And Given an API Schema 'AppTest' related to an Application by post request postAPISchema") {
+			var response = SchemaAPI.createAPISchema(
+				apiApplicationId = ${staticApplicationId},
+				mainObjectDefinitionERC = "L_API_APPLICATION",
+				name = "AppTest");
+		}
+
+		task ("And Given an API EndPoint in an API Application and set Response API Schema to API Endpoints with the created Schema by postAPIEndpoint") {
```
Comment (magjed4289) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[APIRest]APIGenerator/PublishARESTAPIApplication.testcase:46]: not endPoint but endpoint

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[APIRest]APIGenerator/PublishARESTAPIApplication.testcase [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+				method = "getAppTestPage",
+				service = "AppTest");
+		}
+	}
+
+	@priority = 4
+	test CanSeeNewApiEndpointInPublishedApiApplcationPage {
+		property portal.acceptance = "true";
+
+		task ("And Given an API EndPoint 'AppTest' related to an Application by post request postAPIEndpoint") {
```
Comment (magjed4289) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[APIRest]APIGenerator/PublishARESTAPIApplication.testcase:80]: not endPoint but endpoint

**Review by magjed4289 — CHANGES_REQUESTED** [review]

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[APIRest]APIGenerator/PublishARESTAPIApplication.testcase (hunk 3 above)
Comment (Hijky) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[APIRest]APIGenerator/PublishARESTAPIApplication.testcase:80]: thanks, I will modify them

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:headless-functional:nocompile

**Review by magjed4289 — APPROVED** [review]

**Comment by Hijky** [comment]
ci:forward

**Pull request closed without merging**